### PR TITLE
release work for Synse v3

### PR DESCRIPTION
**Pull request by edaniszewski**
This PR:
- merges all the synse v3 work into master (:

### CODE REVIEW

**Comment by lazypower** [comment]
Well, its got a conflict... can you fix that and i'll give it a quick gander?

**Comment by edaniszewski** [comment]
Yup,  just updated. Since all the work in here was previously reviewed before being merged into  the  v3/staging branch, I think just a quick look-over is all that should be needed

File: .jenkins [hunk 1]
```diff
@@ -32,21 +36,32 @@ pipeline {
...
       }
     }
+
+    stage('Python: Publish to PyPi') {
+      when {
+        buildingTag()
+      }
+      environment {
+        TWINE_USERNAME = 'vaporio'
+        TWINE_PASSWORD = credentials('twine-password')
```
Comment (lazypower) [inline .jenkins]: fingers crossed on this again :joy:  i still have memories of the last round of twine. :face_with_head_bandage:

File: python/tests/data/certs/test.key [hunk 2]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (lazypower) [inline python/tests/data/certs/test.key]: whats the story here with these keys? Are they used -anywhere- other than testing and whats the impact of this key getting leaked?

Comment (edaniszewski) [reply python/tests/data/certs/test.key]: They are only being used for tests. Specifically for testing the code paths to configure a secure channel for the client: https://github.com/vapor-ware/synse-server-grpc/blob/v3/staging/python/tests/test_client.py#L59-L67

**Pull request merged**